File: 2025-06-June-GFG-POTD/June 10 - Exactly one swap/Solution.py
class Solution:
    def countStrings(self, s: str) -> int:
        m = [0] * 26
        ans = 0
        for i in range(len(s)):
            ans += i - m[ord(s[i]) - ord('a')]
            m[ord(s[i]) - ord('a')] += 1
        for x in m:
            if x > 1:
                ans += 1
                break
        return ans

File: 2025-06-June-GFG-POTD/June 10 - Exactly one swap/Solution.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int countStrings(string s) {
        vector<int> m(26, 0);
        int ans = 0;
        for (int i = 0; i < s.size(); ++i) {
            ans += i - m[s[i] - 'a'];
            ++m[s[i] - 'a'];
        }
        for (int x : m) {
            if (x > 1) {
                ans++;
                break;
            }
        }
        return ans;
    }
};
